projects/extension/sql/idempotent/012-vectorizer-int.sql

```diff
@@ -805,3 +805,60 @@ $func$
 language plpgsql volatile security invoker
 set search_path to pg_catalog, pg_temp
 ;
+
+-------------------------------------------------------------------------------
+-- _vectorizer_handle_drops
+create or replace function ai._vectorizer_handle_drops()
+returns event_trigger as
+$func$
+declare
+    _vectorizer_id int;
+begin
+    for _vectorizer_id in
+    (
+        select distinct v.id
+        from pg_catalog.pg_event_trigger_dropped_objects() d
+        inner join ai.vectorizer v
+        on ((d.schema_name, d.object_name) in
+            ( (v.source_schema, v.source_table)
+            , (v.target_schema, v.target_table)
+            , (v.queue_schema, v.queue_table)
+            )
+        )
+        where pg_catalog.lower(d.object_type) operator(pg_catalog.=) 'table'
+    )
+    loop
+        -- this may cause recursive invocations of this event trigger
+        -- however it does not cause a problem
+        raise notice 'associated table for vectorizer % dropped. dropping vectorizer', _vectorizer_id;
+        perform ai.drop_vectorizer(_vectorizer_id);
+    end loop;
+end;
+$func$
+language plpgsql volatile security definer
```

Comment:
security definer is required unfortunately

Reply:
You have to be the owner of a table to call `create_vectorizer` on it. When you do, the queue table and target table are created by security invoker functions and thereby owned by the user that owns the source table.

Now that I've slept on it, I'm not sure this needs to be security definer, actually. My thoughts were that we'd need the event trigger to be run as superuser to guarantee that it can drop any tables. I'll try a few things...

Reply:
>Only the table owner, the schema owner, and superuser can drop a table.
https://www.postgresql.org/docs/17/sql-droptable.html

If superuser drops a table, we're good. If table owner drops a table, we're good. If a schema owner drops a table, we may not be okay if the event trigger is `security invoker`. If they own the schema that the source table is in, they likely won't own the `ai` schema where the queue table goes by default.

Reply:
I think maybe we should err on the side of security. I'll make this security invoker, and if the user doesn't have privileges to drop the things, issue a warning and skip